include/perfetto/ext/base/scoped_sched_boost.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef INCLUDE_PERFETTO_EXT_BASE_SCOPED_SCHED_BOOST_H_
+#define INCLUDE_PERFETTO_EXT_BASE_SCOPED_SCHED_BOOST_H_
+
+#include "perfetto/ext/base/status_or.h"
+#include "perfetto/ext/base/thread_checker.h"
+
+namespace perfetto::base {
+
+struct SchedOsConfig {
```

Comment:
why isn't this just an implementation detail?
please don't expose it as it makes things confusing. It seems sorta identical to  SchedPolicyAndPrio